DEVELOPMENT.md

```diff
@@ -188,29 +188,58 @@ Use `go test` to run the integration test suite:
 go test -v ./test/integration -tags=integration
 ```
 
+## Running ko builds with other base images
+By default `ko` uses a lightweight image to run go programs.
+The image can be [overridden](https://ko.build/configuration/#overriding-base-images) with
+the `defaultBaseImage` parameter of the `.ko.yaml` configuration file or through the
+environment variable  `KO_DEFAULTBASEIMAGE`.
+
+An example of an image with a set of debugging tools already installed that can be used is
+`registry.redhat.io/rhel9/support-tools`
+
+**[NOTE]**: Using this images usually needs the pod property `runAsNonRoot` set to `false`.
+
 ## Remote IDE debugging
 
 Use [delve](https://golangforall.com/en/post/go-docker-delve-remote-debug.html)
 to start a debugger to which attach from your IDE.
 
 1. Deploy the chart with `ko` and `helm` in debug mode using an image with `delve`:
+   Specify the [component].debug variable to `true` and update the image accordingly.
+
+   * Deployment to debug the API:
    ```bash
    helm install kubearchive charts/kubearchive --create-namespace -n kubearchive \
    --set apiServer.debug=true \
    --set-string apiServer.image=$(KO_DEFAULTBASEIMAGE=gcr.io/k8s-skaffold/skaffold-debug-support/go:latest ko build --disable-optimizations github.com/kubearchive/kubearchive/cmd/api) \
    --set-string sink.image=$(ko build github.com/kubearchive/kubearchive/cmd/sink) \
    --set-string operator.image=$(ko build github.com/kubearchive/kubearchive/cmd/operator)
    ```
+   * Deployment to debug the Operator:
+   ```bash
+   helm install kubearchive charts/kubearchive --create-namespace -n kubearchive \
+   --set operator.debug=true \
+   --set-string apiServer.image=$(ko build github.com/kubearchive/kubearchive/cmd/api) \
+   --set-string sink.image=$(ko build github.com/kubearchive/kubearchive/cmd/sink) \
+   --set-string operator.image=$(KO_DEFAULTBASEIMAGE=gcr.io/k8s-skaffold/skaffold-debug-support/go:latest ko build --disable-optimizations github.com/kubearchive/kubearchive/cmd/operator)
+   ```
+
+1. Forward the port 40000 from the service that you want to debug:
 
-1. Forward the ports 8081 and 40000 from the Pod directly:
+   * To debug the API we also need the 8081 port for exposing the API Server:
    ```bash
    kubectl port-forward -n kubearchive svc/kubearchive-api-server 8081:8081 40000:40000
    ```
+   * Debug the operator webhooks:
+   ```bash
+   kubectl port-forward -n kubearchive svc/kubearchive-operator-webhooks 40000:40000
+   ```
+
 1. Enable breakpoints in your IDE.
 1. Connect to the process using the port 40000:
     * [VSCode instructions](https://golangforall.com/en/post/go-docker-delve-remote-debug.html#visual-studio-code)
     * [Goland instructions](https://golangforall.com/en/post/go-docker-delve-remote-debug.html#goland-ide)
-1. Query the API using `curl` or your browser:
+1. [Only for the API] Query the API using `curl` or your browser:
```

Comment:
Do we need an instruction about how to generate traffic for the webhooks, the same as we have here for the API?

Reply:
Good point, with the webhooks would be creating a KubeArchiveConfig resource. I'll take care of it

Reply:
On a second thought, we are already creating a KubeArchiveConfig with the test resources, and applying a second resource in a namespace that already has one could be not allowed considering https://github.com/kubearchive/kubearchive/issues/338 so I'll just add a comment without a command